rts/motoko-rts/src/stabilization.rs

```diff
@@ -0,0 +1,74 @@
+//! Graph-copy-based stabilzation on upgrades, serializing the entire stable object graph into
+//! stable memory by using a defined long-term stable storage format.
+//!
+//! This is to support potentially radical changes of the persistent main memory layout, e.g.
+//! introducing a new GC or rearranging persistent metadata. This also relies on precise value
+//! tagging to allow more advanced changes that require value metadata, e.g. specializing arrays for
+//! small element types or even downgrading to 32-bit heap layouts (provided that the amount of live
+//! data fits into a 32-bit memory).
+//!
+//! A memory compatibility check similar to enhanced orthogonal persistence has to be performed.
```

Comment:
I actually wish we did this for the legacy stable format too! But maybe in another life...

Reply:
Also the old candid destabilization is at least somewhat defensive and will catch many errors...

Reply:
I agree, it has different compatibility semantics, and type-driven Candid destabilization is written in a way that it should not lead to corrupt memory state even if the `dfx` incompatibility warning is skipped.